### PR TITLE
Add tetherOptions prop to Popover

### LINKED ISSUE #394: Text blurry in popover when zoomed in Chrome
### Bug report

- __Package version(s)__: 1.2.2
- __Browser and OS versions__: Chrome 54.0.2840.87

#### Steps to reproduce

1. Go to http://blueprintjs.com/docs/#components.menu.js.submenu
2. Zoom the browser window (Ctrl + '+' or Ctrl + mousewheel)

#### Actual behavior

The text in the dropdown menu is blurry

zoom

#### Expected behavior

The text should be crisp

#### Discussion

I've done some research and found that Chrome just doesn't handle subpixel font rendering very well when GPU optimization is enabled. Tether uses the transform attribute to enable GPU optimization for positioning the element but it can be disabled by passing a config option:

````
{
    optimizations: { 
         gpu: false
    }
}
````

There's some discussion here: https://github.com/HubSpot/tether/pull/163 but no definite solution. The simplest solution from a blueprint perspective is to expose the Tether gpu option on the popover component but one might also argue this is something Tether should fix.

What do you guys think?

### PR DESCRIPTION
**Pull request by Binck360**
#### Checklist

- [X] [Enable CircleCI for your fork](https://circleci.com/add-projects)

#### Changes proposed in this pull request:
- Set the constraints prop on Popover to deprecated
- Introduced the tetherOptions prop on Popover, where it is possible to overwrite the settings of Tether

### CODE REVIEW

**Comment by adidahiya** [comment]
We can't remove constraints right now, that's a breaking change. You'll need to handle that as a deprecated prop until 2.0.

File: packages/core/src/components/popover/popover.tsx [hunk 1]
```diff
@@ -52,12 +52,6 @@ export interface IPopoverProps extends IOverlayableProps, IProps {
     arrowSize?: number;
 
     /**
-     * Constraints for the underlying Tether instance.
-     * See http://tether.io/#constraints.
-     */
-    constraints?: TetherUtils.ITetherConstraint[];
```
Comment (giladgray) [inline packages/core/src/components/popover/popover.tsx]: 🔥  unfortunately we cannot remove props until `v2.0`.  please mark this as `@deprecated` in the comment, but you must continue to support it alongside `tetherOptions`.

File: packages/table/src/cell/formats/truncatedFormat.tsx [hunk 2]
```diff
@@ -111,7 +115,7 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, ITru
                     <div className={Classes.TABLE_TRUNCATED_VALUE} ref={this.handleContentDivRef}>{cellContent}</div>
                     <Popover
                         className={Classes.TABLE_TRUNCATED_POPOVER_TARGET}
-                        constraints={constraints}
+                        tetherOptions={tetherOptions}
```
Comment (giladgray) [inline packages/table/src/cell/formats/truncatedFormat.tsx]: how about `tetherOptions={{ constraints }}`? let the typechecker do the work. then you can revert the change above.

File: packages/table/src/headers/columnHeaderCell.tsx [hunk 3]
```diff
@@ -203,15 +203,20 @@ export class ColumnHeaderCell extends React.Component<IColumnHeaderCellProps, IC
...
+                    pin: true,
+                    to: "window",
+                }
+            ]
+        };
 
         return (
             <Popover
-                constraints={constraints}
+                tetherOptions={tetherOptions}
```
Comment (giladgray) [inline packages/table/src/headers/columnHeaderCell.tsx]: same as above.

**Comment by giladgray** [comment]
@Binck360 can you demonstrate how this fixes #394?

File: packages/core/src/components/popover/popover.tsx (hunk 1 above)
Comment (Binck360) [inline packages/core/src/components/popover/popover.tsx]: Oops, didn't stop to think about that. I've reimplemented this again with `@deprecated`

File: packages/table/src/cell/formats/truncatedFormat.tsx (hunk 2 above)
Comment (Binck360) [inline packages/table/src/cell/formats/truncatedFormat.tsx]: Good idea, i've implemented this for truncatedFormat.tsx and columnHeaderCell.tsx. Do I need to do this for contextMenu.tsx too or did you leave this out intentionally?

**Comment by Binck360** [comment]
<h3>Update after linting errors</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/Binck360/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/docs/dist/index.html' target='_blank'>docs</a> | <a href='https://circleci.com/api/v1/project/Binck360/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/Binck360/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/Binck360/blueprint/6/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

File: packages/table/src/cell/formats/truncatedFormat.tsx [hunk 4]
```diff
@@ -112,6 +112,7 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, ITru
                     <Popover
                         className={Classes.TABLE_TRUNCATED_POPOVER_TARGET}
                         constraints={constraints}
```
Comment (adidahiya) [inline packages/table/src/cell/formats/truncatedFormat.tsx]: you don't need this line anymore, actually would prefer removing it since it's `@deprecated`

File: packages/table/src/headers/columnHeaderCell.tsx [hunk 5]
```diff
@@ -212,6 +213,7 @@ export class ColumnHeaderCell extends React.Component<IColumnHeaderCellProps, IC
         return (
             <Popover
                 constraints={constraints}
```
Comment (adidahiya) [inline packages/table/src/headers/columnHeaderCell.tsx]: don't need this prop either

File: packages/core/src/components/popover/popover.tsx [hunk 6]
```diff
@@ -514,9 +521,16 @@ export class Popover extends AbstractComponent<IPopoverProps, IPopoverState> {
             // so instead, we'll position tether based off of its first child.
             // NOTE: use findDOMNode(this) directly because this.targetElement may not exist yet
             const target = findDOMNode(this).childNodes[0];
+            let tetherUtilOptions = this.props.tetherOptions || {};
```
Comment (adidahiya) [inline packages/core/src/components/popover/popover.tsx]: can you supply this default with `defaultProps` instead? then you don't need any `tetherUtilOptions` variable. also, we generally prefer to avoid implicit type coercion (here, `this.props.tetherOptions` is coerced to a boolean)

File: packages/table/src/cell/formats/truncatedFormat.tsx [hunk 7]
```diff
@@ -112,6 +112,7 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, ITru
...
+                        tetherOptions={{constraints}}
```
Comment (giladgray) [inline packages/table/src/cell/formats/truncatedFormat.tsx]: convention is spaces in braces for objects `{{ constraints }}`

**Comment by giladgray** [comment]
🌮 looking good! please make the changes noted above and we'll ship this in tomorrow's release.

**Comment by Binck360** [comment]
<h3>Sorted tetherOptions alphabetically</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/Binck360/blueprint/8/artifacts/0//home/ubuntu/blueprint/packages/docs/dist/index.html' target='_blank'>docs</a> | <a href='https://circleci.com/api/v1/project/Binck360/blueprint/8/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/Binck360/blueprint/8/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/Binck360/blueprint/8/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

File: packages/core/src/components/popover/popover.tsx [hunk 8]
```diff
@@ -514,9 +522,15 @@ export class Popover extends AbstractComponent<IPopoverProps, IPopoverState> {
             // so instead, we'll position tether based off of its first child.
             // NOTE: use findDOMNode(this) directly because this.targetElement may not exist yet
             const target = findDOMNode(this).childNodes[0];
+
+            // This is for backwards compatibility. this.props.constraints is always leading
```
Comment (giladgray) [inline packages/core/src/components/popover/popover.tsx]: not a full sentence. always leading what?

**Comment by giladgray** [comment]
thanks @Binck360! 🎩
